test/CodeGenerator/AbstractInjectorTest.php

```diff
@@ -0,0 +1,197 @@
+            {
+                $this->factories = ($this->provider)();
+            }
+        };
+    }
+
+    public function testImplementsContract()
+    {
+        $prophecy = $this->prophesize(InvokableInterface::class);
+        $prophecy->__call('__invoke', [])
+            ->shouldBeCalled()
+            ->willReturn([
+                'SomeService' => 'SomeFactory'
+            ]);
+
+        $subject = $this->createTestSubject($prophecy->reveal());
+        $this->assertInstanceOf(InjectorInterface::class, $subject);
+    }
+
+    public function testCanCreateReturnsTrueWhenAFactoryIsAvailable()
+    {
+        $className = uniqid('SomeClass');
+        $provider = function () use ($className) {
+            return [$className => 'SomeClassFactory'];
+        };
+
+        $this->decoratedInjectorProphecy
+            ->canCreate($className)
+            ->shouldNotBeCalled();
+
+        $subject = $this->createTestSubject($provider);
+        $this->assertTrue($subject->canCreate($className));
+    }
+
+    public function testCanCreateUsesDecoratedInjectorWithoutFactory()
+    {
+        $missingClass = uniqid('SomeClass');
+        $existingClass = uniqid('SomeOtherClass');
+        $provider = function () {
+            return [];
+        };
+
+        $this->decoratedInjectorProphecy
+            ->canCreate($missingClass)
+            ->shouldBeCalled()
+            ->willReturn(false);
+
+        $this->decoratedInjectorProphecy
+            ->canCreate($existingClass)
+            ->shouldBeCalled()
+            ->willReturn(true);
+
+        $subject = $this->createTestSubject($provider);
+
+        $this->assertTrue($subject->canCreate($existingClass));
+        $this->assertFalse($subject->canCreate($missingClass));
+    }
+
+    public function testCreateUsesFactory()
+    {
+        $factory = $this->prophesize(FactoryInterface::class);
+        $className = uniqid('SomeClass');
+        $params = ['someArg' => uniqid()];
+        $expected = new stdClass();
+        $provider = function () use ($className, $factory) {
+            return [$className => $factory->reveal()];
+        };
+
+        $factory->create(
+            $this->containerProphecy->reveal(),
+            $params
+        )
+            ->shouldBeCalled()
+            ->willReturn($expected);
+
+        $this->decoratedInjectorProphecy
+            ->create($className, Argument::cetera())
+            ->shouldNotBeCalled();
+
+        $subject = $this->createTestSubject($provider);
+        $this->assertSame($expected, $subject->create($className, $params));
+    }
+
+    public function testCreateUsesDecoratedInjectorIfNoFactoryIsAvailable()
+    {
+        $className = uniqid('SomeClass');
+        $expected = new stdClass();
+        $params = [ 'someArg' => uniqid() ];
+        $provider = function () {
+            return [];
+        };
+
+        $this->decoratedInjectorProphecy->create($className, $params)
```

Comment:
Move the `->create()` call to the next line. When flowing to multiple lines, arrows should be assumed to be acting on whatever the return/lookup of the first line was.